instaclone/src/App.js

```diff
@@ -5,23 +5,61 @@ import './App.css';
 import SearchBar from './components/SearchBar/SearchBar';
 
 class App extends React.Component {
-  state={
-    data: dummyData
+  constructor(){
+    super();
+    this.state={
+    data: [],
+    filter: '',
+    searchInput: ''
+    }
   }
+  
+  componentDidMount() {
+    this.setState({ data: dummyData });
+    
+  }
+
+  
 
+  searchTerm = e => {
+    
+     this.setState({ searchInput: e.target.value });
+  };
 
+    submithandle = (e) => {
```

Comment:
The `submithandle` still not working properly here because you only filter the data inside this function without updating the data that you render.